src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -494,83 +416,80 @@ private IEnumerable<Claim> OutboundClaimTypeTransform(IEnumerable<Claim> claims)
             }
         }
 
+        public JwtSecurityToken ReadJwtToken(string token)
+        {
+            if (token == null)
+                throw LogHelper.LogArgumentNullException("token");
+
+            if (token.Length * 2 > MaximumTokenSizeInBytes)
+                throw LogHelper.LogException<ArgumentException>(LogMessages.IDX10209, token.Length, MaximumTokenSizeInBytes);
+
+            if (!CanReadToken(token))
+                throw LogHelper.LogException<ArgumentException>(LogMessages.IDX10708, GetType(), token);
+
+            var jwt = new JwtSecurityToken();
+            jwt.Decode(token);
+            return jwt;
```

Comment:
I didn't do that because, I didn't want to do all the checks again in the ctor of JwtSecurityToken.